runtime/vm/resolvefield.cpp

```diff
@@ -634,19 +634,21 @@ UDATA
 findIndexInFlattenedClassCache(J9FlattenedClassCache *flattenedClassCache, J9ROMNameAndSignature *nameAndSignature)
 {
 	/* first field indicates the number of classes in the cache */
-	UDATA length = flattenedClassCache->offset;
+	UDATA length = flattenedClassCache->numberOfEntries;
 	UDATA index = 0;
+	bool found = false;
 
-	for (UDATA i = 1; i <= length; i++) {
-		if (J9UTF8_EQUALS(J9ROMNAMEANDSIGNATURE_NAME(nameAndSignature), J9ROMNAMEANDSIGNATURE_NAME(flattenedClassCache[i].nameAndSignature))
-			&& J9UTF8_EQUALS(J9ROMNAMEANDSIGNATURE_SIGNATURE(nameAndSignature), J9ROMNAMEANDSIGNATURE_SIGNATURE(flattenedClassCache[i].nameAndSignature))
+	for (UDATA i = 0; i < length; i++) {
+		if (J9UTF8_EQUALS(J9ROMNAMEANDSIGNATURE_NAME(nameAndSignature), J9ROMNAMEANDSIGNATURE_NAME(flattenedClassCache->entry[i].nameAndSignature))
+			&& J9UTF8_EQUALS(J9ROMNAMEANDSIGNATURE_SIGNATURE(nameAndSignature), J9ROMNAMEANDSIGNATURE_SIGNATURE(flattenedClassCache->entry[i].nameAndSignature))
 		) {
 			index = i;
+			found = true;
 			break;
 		}
 	}
 
-	Assert_VM_true(index != 0);
+	Assert_VM_true(found);
```

Comment:
you dont need this assertion, this was only done to catch if we return a header field. With the restructuring this is no longer possible